src/libstore/build.cc

```diff
@@ -1875,6 +1878,28 @@ void DerivationGoal::startBuilder()
             useChroot = false;
         else if (settings.sandboxMode == smRelaxed)
             useChroot = !fixedOutput && !noChroot;
+        else if (settings.sandboxMode == smEnabledIfAvailable) {
+            useChroot = true;
+#if __linux__
+            // Enable sandboxing only if the system supports it. A
+            // clone is tried to determine if it works. Success
+            // indicates sandboxing can be enabled, otherwise disable
+            // sandboxing
+
+            int flags = CLONE_NEWUSER | CLONE_NEWNS | CLONE_NEWIPC | CLONE_NEWUTS | CLONE_PARENT | SIGCHLD;
+            size_t stackSize = 1;
+            char* stack = malloc(stackSize);
```

Comment:
Probably better to use `std::make_unique<unsigned char[]>(stackSize);`. Leaked mallocs pollute valgrind runs.